### PR TITLE
Ros2 interface

### PR DESCRIPTION
**Pull request by Ryanf55**
# Purpose

Start some documentation on the ROS interface on which channels are exposed. This also corrects a few related issues in the wiki docs.

# Related PR's

This Wiki is up to date assuming the following PR's all get merged.

* https://github.com/ArduPilot/ardupilot/pull/26911
* https://github.com/ArduPilot/ardupilot/pull/28401
* https://github.com/ArduPilot/ardupilot/pull/28434
* https://github.com/ArduPilot/ardupilot/pull/28441

### CODE REVIEW

**Comment by Ryanf55** [comment]
@pedro-fuoco  Can you review/add information to the TF , localization, and odometry sections? I'd like to make sure it has the info that your project figured out for GPS denied navigation.

File: dev/source/docs/ros2-interfaces.rst
Comment (pedro-fuoco) [reply dev/source/docs/ros2-interfaces.rst:249]: Sidenote: This shouldn't stop this PR from being approved

**Comment by pedro-fuoco** [comment]
I will add information to the sections you requested tomorrow

File: dev/source/docs/ros2-interfaces.rst
Comment (Ryanf55) [reply dev/source/docs/ros2-interfaces.rst:249]: Let's make sure we have CI running the existing tests for jazzy before merge.

**Review by pedro-fuoco — CHANGES_REQUESTED** [review]
Do the services not have a table describing their types because they are using custom ardupilot types ?

**Review by pedro-fuoco — CHANGES_REQUESTED** [review]
Let me know what you think about the additions I proposed. It would be best if the merge waited until `/ap/pose/filtered` is documented

**Comment by Ryanf55** [comment]
> Do the services not have a table describing their types because they are using custom ardupilot types ?

I just didn't have time to document them yet, Perhaps in a follow up they could be added.

**Review by pedro-fuoco — APPROVED** [review]
LGTM, we can merge once the PR dependencies are done.
Thanks for the contribution @Ryanf55!

**Comment by Ryanf55** [comment]
@Hwurzburg Please merge when you are happy. All dependencies are now merged.

**Comment by Hwurzburg** [comment]
tnx

**Pull request merged**
